exercises/practice/rational-numbers/test/rational_numbers_test.exs

```diff
@@ -0,0 +1,313 @@
+      r1 = {1, 2}
+      r2 = {-2, 3}
+      result = {-3, 4}
+      assert RationalNumbers.divide_by(r1, r2) == result
+    end
+
+    @tag :pending
+    test "Divide two negative rational numbers" do
+      r1 = {-1, 2}
+      r2 = {-2, 3}
+      result = {3, 4}
+      assert RationalNumbers.divide_by(r1, r2) == result
+    end
+
+    @tag :pending
+    test "Divide a rational number by 1" do
+      r1 = {1, 2}
+      r2 = {1, 1}
+      result = {1, 2}
+      assert RationalNumbers.divide_by(r1, r2) == result
+    end
+  end
+
+  describe "Absolute value" do
+    @tag :pending
+    test "Absolute value of a positive rational number" do
+      r = {1, 2}
+      result = {1, 2}
+      assert RationalNumbers.abs(r) == result
+    end
+
+    @tag :pending
+    test "Absolute value of a positive rational number with negative numerator and denominator" do
+      r = {-1, -2}
+      result = {1, 2}
+      assert RationalNumbers.abs(r) == result
+    end
+
+    @tag :pending
+    test "Absolute value of a negative rational number" do
+      r = {-1, 2}
+      result = {1, 2}
+      assert RationalNumbers.abs(r) == result
+    end
+
+    @tag :pending
+    test "Absolute value of a negative rational number with negative denominator" do
+      r = {1, -2}
+      result = {1, 2}
+      assert RationalNumbers.abs(r) == result
+    end
+
+    @tag :pending
+    test "Absolute value of zero" do
+      r = {0, 1}
+      result = {0, 1}
+      assert RationalNumbers.abs(r) == result
+    end
+  end
+
+  describe "Exponentiation of a rational number" do
+    @tag :pending
+    test "Raise a positive rational number to a positive integer power" do
+      r = {1, 2}
+      n = 3
+      result = {1, 8}
+      assert RationalNumbers.pow_rational(r, n) == result
+    end
+
+    @tag :pending
+    test "Raise a negative rational number to a positive integer power" do
+      r = {-1, 2}
+      n = 3
+      result = {-1, 8}
+      assert RationalNumbers.pow_rational(r, n) == result
+    end
+
+    @tag :pending
+    test "Raise zero to an integer power" do
+      r = {0, 1}
+      n = 5
+      result = {0, 1}
+      assert RationalNumbers.pow_rational(r, n) == result
+    end
+
+    @tag :pending
+    test "Raise one to an integer power" do
+      r = {1, 1}
+      n = 4
+      result = {1, 1}
+      assert RationalNumbers.pow_rational(r, n) == result
+    end
+
+    @tag :pending
+    test "Raise a positive rational number to the power of zero" do
+      r = {1, 2}
+      n = 0
+      result = {1, 1}
+      assert RationalNumbers.pow_rational(r, n) == result
+    end
+
+    @tag :pending
+    test "Raise a negative rational number to the power of zero" do
+      r = {-1, 2}
+      n = 0
+      result = {1, 1}
+      assert RationalNumbers.pow_rational(r, n) == result
+    end
+  end
+
+  describe "Exponentiation of a real number to a rational number" do
+    # We round float values to 10 decimal places to allow comparaison
```

Comment:
Let's use [`assert_in_delta`](https://hexdocs.pm/ex_unit/ExUnit.Assertions.html#assert_in_delta/4) instead of rounding. It's the usual tool to use in unit tests for floats 🙂. I'm not certain if it makes any practical difference, but at least the code will be a bit shorter. Maybe we can put the delta value in a module attribute?

Reply:
Thank you, I didn't know about `assert_in_delta` even though in hindsight of course it exists.